python/fledge/services/core/api/control_service/script_management.py

```diff
@@ -30,14 +30,144 @@
     -----------------------------------------------------------------------
     | GET POST            | /fledge/control/script                        |
     | GET PUT DELETE      | /fledge/control/script/{script_name}          |
+    | GET                 | /fledge/control/script/schedule               |
     | POST                | /fledge/control/script/{script_name}/schedule |
     -----------------------------------------------------------------------
 """
 
 _logger = logger.setup(__name__, level=logging.INFO)
 
 
-async def get_all_scripts(request: web.Request) -> web.Response:
+def setup(app):
+    # schedules
+    app.router.add_route('GET', '/fledge/control/script/schedule', get_all_schedules)
+    app.router.add_route('GET', '/fledge/control/script/{script_name}/schedule', get_schedule_by_name)
+    app.router.add_route('POST', '/fledge/control/script/{script_name}/schedule', add_schedule_and_configuration)
+
+    # CRUD's
+    app.router.add_route('POST', '/fledge/control/script', add)
+    app.router.add_route('GET', '/fledge/control/script', get_all)
+    app.router.add_route('GET', '/fledge/control/script/{script_name}', get_by_name)
+    app.router.add_route('PUT', '/fledge/control/script/{script_name}', update)
+    app.router.add_route('DELETE', '/fledge/control/script/{script_name}', delete)
+
+
+async def get_all_schedules(request: web.Request) -> web.Response:
+    """ Get list of automation script type schedule
+
+    :Example:
+        curl -H "authorization: $AUTH_TOKEN" -sX GET http://localhost:8081/fledge/control/script/schedule
+    """
+    schedule_list = await server.Server.scheduler.get_schedules()
+    schedules = []
+    for sch in schedule_list:
+        if sch.process_name == "automation_script":
+            schedules.append({
+                'id': str(sch.schedule_id),
+                'name': sch.name,
+                'processName': sch.process_name,
+                'type': Schedule.Type(int(sch.schedule_type)).name,
+                'repeat': 0,
+                'time': 0,
+                'day': sch.day,
+                'exclusive': sch.exclusive,
+                'enabled': sch.enabled
```

Comment:
It should return not only the schedules but also the config data i.e. parameter. In GUI we can always filter these with the process_name. To avoid the multiple calls to fetch the resource details by the client we should have the CRUD operations with schedule and configuration data.

Reply:
@MarkRiddoch 
Please review the requirement and document itself in control service support REST API doc.
IMHO, We may not require these new endpoints, we can use the existing endpoint and add the schedule & category info inside it.